### PR TITLE
Fix operator oom

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:
Keep the latest used install strategy.

The current cache behavior is to cache every
install strategy and only clear the cache when
Kubevirt is uninstalled.

This is increasing memory consumption with every change
to Kubevirt CR and ultimately lead to oom.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-operator/kubevirt.go [hunk 1]
```diff
@@ -820,18 +824,23 @@ func (c *KubeVirtController) garbageCollectInstallStrategyJobs() error {
...
+	}
+	return nil, false
 }
 
-func (c *KubeVirtController) cacheInstallStrategyInMap(strategy *install.Strategy, config *operatorutil.KubeVirtDeploymentConfig, generation int64) {
-
-	c.installStrategyMutex.Lock()
-	defer c.installStrategyMutex.Unlock()
-	c.installStrategyMap[fmt.Sprintf(installStrategyKeyTemplate, config.GetDeploymentID(), generation)] = strategy
+func (c *KubeVirtController) cacheInstallStrategyInMap(cachedEntry *install.Strategy, config *operatorutil.KubeVirtDeploymentConfig, generation int64) {
```
Comment (iholder101) [inline pkg/virt-operator/kubevirt.go]: nit: maybe we can also change the function name?

**Review by iholder101 — COMMENTED** [review]
Looks great!
/lgtm

File: pkg/virt-operator/kubevirt.go (hunk 1 above)
Comment (xpivarc) [reply pkg/virt-operator/kubevirt.go]: See the latest commit.

**Comment by vladikr** [comment]
Sorry for the delay. It took me some time to wrap my head around this change.
/approve

**Comment by xpivarc** [comment]
> Sorry for the delay. It took me some time to wrap my head around this change. /approve

Anything that would help? :)

**Comment by xpivarc** [comment]
> Sorry for the delay. It took me some time to wrap my head around this change. /approve

Anything that would help? :)

**Pull request merged**